sparse/mlir_backend/formats.py

```diff
@@ -61,7 +61,7 @@ def build(self):
 
 
 @dataclasses.dataclass(eq=True, frozen=True, kw_only=True)
-class StorageFormat:
+class ConcreteFormat:
```

Comment:
`ConcreteFormat` suggests that there's  `AbstractFormat` somewhere. Didn't `StorageFormat` fit here? Why is our format concrete now?

Reply:
A format factory is what most users would think of as a format (just specify the format, ndim and dtype, and be ready to go), whereas `StorageFormat` is more concrete: It includes pos_width and crd_width as well as very precise levels and order. Additionally, it is what one would allocate as an output.